canvas/services/event.py

```diff
@@ -52,6 +44,29 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    users = []
+    progress_submission_statuses = [0, 0, 0]
```

Comment:
This variable actually stores counts of students that correspond to various submission statuses, this variable name feels iffy based on this description.

Second ick is that it's a bit messy to understand which index corresponds to each status (but reading the code makes it fairly clear tbh) - still I'd probably have used a dictionary with keys that describe what the value is.
i.e. (kinda weak naming on my part here, adjust as makes sense)
```py
best_submission_counts = {
    "num_correct": 0,
    "num_partially_correct": 0,
    "num_incorrect": 0,
}
```

Especially since when we return this value we are returning a dictionary that names each of them anyway